pandas/tseries/resample.py

```diff
@@ -323,6 +323,11 @@ def aggregate(self, arg, *args, **kwargs):
                                                *args,
                                                **kwargs)
 
+        # if arg was a string, _aggregate called resampler's _downsample or
+        # _groupby_and_agg methods, which would've already applied the loffset
+        if not isinstance(arg, compat.string_types):
```

Comment:
It's not happening in there because `.downsample` is not called when anything other than a string is passed into `agg`. 

`Resampler` has downsample & groupby_and_agg methods (e.g. count, mean) which in return call `_downsample`, and when a string is passed into `agg`, these methods are called. However, if a string isn't passed in, these methods would not have been called, and consequently `_downsample` would not have been called.